modules/core/src/main/java/io/fluo/core/metrics/MetricsUtil.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2014 Fluo authors (see AUTHORS)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.fluo.core.metrics;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Reservoir;
+import com.codahale.metrics.Timer;
+import io.fluo.api.config.FluoConfiguration;
+import io.fluo.core.impl.FluoConfigurationImpl;
+import io.fluo.core.worker.NotificationFinder;
+import io.fluo.core.worker.finder.hash.HashNotificationFinder;
+import org.mpierce.metrics.reservoir.hdrhistogram.HdrHistogramResetOnSnapshotReservoir;
+
+public class MetricsUtil {
+
+  public static Reservoir getConfiguredReservoir(FluoConfiguration config) {
+    String clazz =
+        config.getString(FluoConfigurationImpl.METRICS_RESERVOIR_PROP,
+            HdrHistogramResetOnSnapshotReservoir.class.getName());
+    try {
+      return Class.forName(clazz).asSubclass(Reservoir.class).newInstance();
+    } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  public static Timer getTimer(FluoConfiguration config, MetricRegistry registry, String name) {
+    Timer timer = registry.getTimers().get(name);
+    if (timer == null) {
+      timer = new Timer(getConfiguredReservoir(config));
+      registry.register(name, timer);
```

Comment:
there is a concurrency issue here... two threads could call this and both try to register... I looked at the register code, seems to throw an exception if name is already registered